compact.go

```diff
@@ -360,16 +384,20 @@ func (w *instrumentedChunkWriter) WriteChunks(chunks ...ChunkMeta) error {
 func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockReader) (err error) {
 	level.Info(c.logger).Log("msg", "compact blocks", "count", len(blocks), "mint", meta.MinTime, "maxt", meta.MaxTime)
 
+	var tmp string
+
```

Comment:
Maybe just move the defer below setting of `dir` and `tmp`.

Reply:
Done.